pkg/validation/limits.go

```diff
@@ -234,7 +235,16 @@ func (l *Limits) Validate() error {
 			l.StreamRetention[i].Matchers = matchers
 		}
 	}
-	return nil
+
+	err := fmt.Errorf("delete format must be one of 'disabled', 'filter-only', or 'filter-and-delete'")
```

Comment:
Ideally we don't want this to be just strings, but importing `deletion` creates an import loop that's not trivial to unravel. If we want it, I'm happy to make a follow in PR

Reply:
I'm afraid of how easy this is to break when we add new deletion modes but forget to update it here. It should be fairly easy to add a new deletionmode package that just has the `DeletionMode` types we need and can be imported.